internal/docstore/firedocstore/codec.go

```diff
@@ -70,31 +70,41 @@ type encoder struct {
 
 var nullValue = &pb.Value{ValueType: &pb.Value_NullValue{}}
 
-func (e *encoder) EncodeNil()            { e.pv = nullValue }
-func (e *encoder) EncodeBool(x bool)     { e.pv = &pb.Value{ValueType: &pb.Value_BooleanValue{x}} }
-func (e *encoder) EncodeInt(x int64)     { e.pv = &pb.Value{ValueType: &pb.Value_IntegerValue{x}} }
-func (e *encoder) EncodeUint(x uint64)   { e.pv = &pb.Value{ValueType: &pb.Value_IntegerValue{int64(x)}} }
-func (e *encoder) EncodeBytes(x []byte)  { e.pv = &pb.Value{ValueType: &pb.Value_BytesValue{x}} }
+func (e *encoder) EncodeNil() { e.pv = nullValue }
+func (e *encoder) EncodeBool(x bool) {
+	e.pv = &pb.Value{ValueType: &pb.Value_BooleanValue{BooleanValue: x}}
```

Comment:
This is a known vet bug/misfeature: proto oneof structs will only ever have one field by design, so this isn't necessary.

Reply:
@dsnet What's the current status of `go vet` vs. proto oneof structs? Is there an issue somewhere that discusses this?

Reply:
I'm not aware of any discussion about this, but your earlier statement that "oneof structs will only ever have one field by design" is pretty much true.

Reply:
@vangent can you remove the struct keys from the oneofs? I'm okay with everything else.

Reply:
Done. I left the `&pb.Precondition_Exists{Exists: true}`, let me know if you think I should drop those too.